src/climate_indices/compute.py

```diff
@@ -412,8 +412,7 @@ def _pearson_fit(
         # that threshold then we set the corresponding CDF to a floor value
         # TODO ask Richard Heim why the use of this floor value, matching
         #  that used for the trace amount?
-        nans_mask = np.isnan(values)
-        values[np.logical_and(minimums_mask, nans_mask)] = 0.0005
+        values[minimums_mask] = 0.0005
 
         # account for negative skew
         skew_mask = skew < 0.0
```

Comment:
Just had an inspiration about where we can find more authoritative information. I went back to the GPL'ed library I was originally looking at to see what sort of references they were using without looking at their code. This PDF looks promising: https://github.com/e-baumer/standard_precip/blob/master/docs/lloyd_hughes.pdf

Reply:
Yes, that paper indicates that their implementation of the gamma distribution fitting worked best for Europe. [Guttman (1998)](https://drought.unl.edu/archive/Documents/NDMC/Workshops/136/Ref/SPI_Guttman.pdf) votes in favor of the Pearson III for the USA, not because it's any better than gamma but because it has more parameters:  
```
Because the PE3 with its three parameters is more flexible than the GAM with its two parameters in fitting sample data, it is suggested that the PE3 be used as a universal model from which to compute the SPI from site data. Although not discussed, the PE3 model is also the best choice for describing wet events.
```
Since Guttman worked in the same office they went with his suggestion, and I eventually inherited the code when I was hired years later. There were numerous versions in play at NOAA and several others in the wild, each one computing things slightly differently with varying measures of agreement across the spectrum, and I thought this shouldn't be, there should instead be a community reference implementation that everyone can converge on as the right way to compute SPI (and the other indices introduced later), then there can be less re-invention of the wheel and more standardization of calculations. With a commonly used open-source package then studies have a much better chance at being reproducible, unlike the two papers we've referenced above -- I'll bet dollars to doughnuts it can't be done for either one since the code is lost to history. But I digress...

Reply:
Learning lots of good stuff from my "Statistical Methods in the Atmospheric Sciences" book, or at least, good clarifying information. Still haven't completely cracked this nut, but one possible theory I am mulling in my head is that the negative skew handling may have been a misguided attempt at making the PDF artificially handle negative skews by making it symmetric w.r.t. skew. I have found another application of gamma distributions where something like this is done, but not for skew. The pearson III distribution already allows for negative skew, so it just makes me think the code is incorrect. But I am still not sure.

Reply:
Alright, I have been doing some plotting, and without knowing exactly the rationale _for_ doing the `1 - values` when the skew is negative, I am now convinced that I could never come up with my own justification for doing so. And just to show, for the test data that is causing the failure, here is its histogram and its fitted pearson3 distribution.
![image](https://user-images.githubusercontent.com/291576/139728858-dd7536bd-caad-470d-befc-20042c26234e.png)

And here is what the transformed values look like with and without the "negative skew modification". Note that the blue points (no modification for negative skew) look a lot like the transformed values for the months with positive skews.
![image](https://user-images.githubusercontent.com/291576/139729644-e46bcde0-5297-4cbe-b065-b2ccec5dcec7.png)

Reply:
Ah! I think I found a reason! https://github.com/scipy/scipy/issues/11186

There was a bug in earlier versions of scipy that caused the cdf to be incorrect for negative skews! This was only fixed in the past year, it looks like!

Reply:
Also looks like the same set of fixes also fixed an issue that might have involved the nan mask, too. Maybe this means we need a scipy version check to decide what to do.

Reply:
Let this be a lesson: when working around a bug in another library, file a bug report and include a link to that bug report in the comments for the work-around.

Reply:
Agreed, that's the way. For what it's worth we never knew there was a bug in scipy causing this work-around to be baked into our code, so when it was fixed we were caught with our pants down.

Reply:
Not completely. Your unit test did flag something, which is better than nothing. If you didn't have that one timeseries in the test data that just so happened to skew negatively, then your pants would have been down...